### PR TITLE
Document our support policy for Elastic

### PR DESCRIPTION
**Pull request by tomwilkie**
**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/datasources/elasticsearch/_index.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ This data source supports these versions of Elasticsearch:
 - v7.16+
 - v8.x
 
+Our policy is to follow [Elastic Product End of Life Dates](https://www.elastic.co/support/eol) to reduce maintenance burden.  Practically this means you will get a notification in the query builder that the version of Elastic you are using is no longer supported - nothing will prevent the queries from running.
+
```
Comment (ivanahuckova) [inline docs/sources/datasources/elasticsearch/_index.md:37]:
```suggestion
Our maintenance policy for Elasticsearch data source is aligned with the [Elastic Product End of Life Dates](https://www.elastic.co/support/eol) and we ensure proper functionality for supported versions. If you are using an Elasticsearch with version that is past its end-of-life (EOL), you can still execute queries, but you will receive a notification in the query builder indicating that the version of Elasticsearch you are using is no longer supported. It's important to note that in such cases, we do not guarantee the correctness of the functionality, and we will not be addressing any related issues.
```

**Comment by ivanahuckova** [comment]
I have updated the content. @gabor @Eve832 would you please review changes.

**Review by gabor — APPROVED** [review]
LGTM 👍

**Comment by tomwilkie** [comment]
Thank you @ivanahuckova!

**Pull request merged**